backend/src/chat.gateway.ts

```diff
@@ -0,0 +1,170 @@
+import { Logger } from '@nestjs/common';
+import {
+  OnGatewayConnection,
+  OnGatewayDisconnect,
+  OnGatewayInit,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+  WsException,
+} from '@nestjs/websockets';
+import { Socket, Server } from 'socket.io';
+import { CreateMessageDto } from './messages/dto/create-message.dto';
+import { MessagesService } from './messages/messages.service';
+import * as jwt from 'jsonwebtoken';
+import { UsersService } from './users/users.service';
+import { User } from './users/entities/user.entity';
+import { TokenUserDto } from './auth/dto/token-user.dto';
+import { plainToInstance } from 'class-transformer';
+import { ResponseMessageDto } from './messages/dto/response-message.dto';
+import { MembershipsService } from './memberships/memberships.service';
+import { ConfigService } from '@nestjs/config';
+import { MembershipRoleType } from './memberships/entities/membership.entity';
+import { CreateMembershipDto } from './memberships/dto/create-membership.dto';
+
+export class ChatJoinDto {
+  channel: string;
+}
+
+export class ChatSendDto {
+  channel: string;
+  message: string;
+}
+
+@WebSocketGateway({ cors: true })
```

Comment:
```suggestion
@WebSocketGateway({ cors: true, namespace: 'chat' })
```
This would only require from the client to connect to `ws://localhost:8080/chat` in place of `ws://localhost:8080`.

Reply:
I think it is better to only have one ws connection for the whole app, no?

Reply:
In fact, there is only one TCP connection, the namespace are virtual, see [this article](https://coloringchaos.github.io/rtw-s17/namespaces-rooms/).

Reply:
Okay, great, thank you for the article. I have been trying to use Vue's `socket.io`  package, which makes the connection when the app loads. But using socket.io directly and using these namespaces seems much better. It will let us isolate the chat from the game, and potentially other notifications if necessary.